runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecord.java

```diff
@@ -43,83 +44,107 @@
+    @Setter
+    long globalAddress = Address.NON_ADDRESS;
+
+    /**
+     * Set the upcall result for this entry.
+     */
     public void setUpcallResult(Object result) {
         upcallResult = result;
         haveUpcallResult = true;
     }
 
-    /** Set the undo record for this entry. */
+    /**
+     * Set the undo record for this entry.
+     */
     public void setUndoRecord(Object object) {
         this.undoRecord = object;
         undoable = true;
     }
 
-    /** Clear the undo record for this entry. */
+    /**
+     * Clear the undo record for this entry.
+     */
     public void clearUndoRecord() {
         this.undoRecord = null;
         undoable = false;
     }
 
 
-    /** SMREntry constructor. */
-    public SMREntry(String smrMethod, @NonNull Object[] smrArguments, ISerializer serializer) {
-        super(LogEntryType.SMR);
+    /**
+     * SMRRecord constructor.
+     */
+    public SMRRecord(String smrMethod, @NonNull Object[] smrArguments, ISerializer serializer) {
         this.SMRMethod = smrMethod;
         this.SMRArguments = smrArguments;
         this.serializerType = serializer;
     }
 
     /**
-     * This function provides the remaining buffer. Child entries
-     * should initialize their contents based on the buffer.
+     * Deserialize from buffer and return a new {@code SMRRecord}.
      *
-     * @param b The remaining buffer.
+     * @param b byte buffer to deserialize.
+     * @return an {@code SMRRecord} deserialized from buffer.
      */
-    @Override
-    void deserializeBuffer(ByteBuf b, CorfuRuntime rt) {
-        super.deserializeBuffer(b, rt);
+    static SMRRecord deserializeFromBuffer(ByteBuf b, CorfuRuntime rt) {
+        SMRRecord record = new SMRRecord();
         short methodLength = b.readShort();
         byte[] methodBytes = new byte[methodLength];
         b.readBytes(methodBytes, 0, methodLength);
-        SMRMethod = new String(methodBytes);
-        serializerType = Serializers.getSerializer(b.readByte());
+        record.SMRMethod = new String(methodBytes, DEFAULT_CHAR_SET);
+        record.serializerType = Serializers.getSerializer(b.readByte());
         byte numArguments = b.readByte();
         Object[] arguments = new Object[numArguments];
         for (byte arg = 0; arg < numArguments; arg++) {
             int len = b.readInt();
             ByteBuf objBuf = b.slice(b.readerIndex(), len);
-            arguments[arg] = serializerType.deserialize(objBuf, rt);
+            arguments[arg] = record.serializerType.deserialize(objBuf, rt);
             b.skipBytes(len);
         }
-        SMRArguments = arguments;
+        record.SMRArguments = arguments;
+
+        return record;
     }
 
-    @Override
-    public void serialize(ByteBuf b) {
-        super.serialize(b);
+    /**
+     * Serialize this record into a buffer.
+     * The global address is not serialized.
+     *
+     * @param b byte buffer to serialize to.
+     */
+    void serialize(ByteBuf b) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651468)